scripts/install/install-helper-functions.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash 
+# This file is to be sourced into another to add helper functions, rather than
+# executed directly.
+
+#-------------------------------------------------------------------------------
+# Logging support (nascent; I want to add more control and a log file).
+
+# Print a separator bar of # characters.
+function echo_bar() {
+  local terminal_width
+  if [ -n "${TERM}" ] && [ -t 0 ] ; then
+    if [[ -n "$(which resize)" ]] ; then
+      terminal_width="$(resize 2>/dev/null | grep COLUMNS= | cut -d= -f2)"
+    elif [[ -n "$(which stty)" ]] ; then
+      terminal_width="$(stty size 2>/dev/null | cut '-d ' -f2)"
+    fi
+  fi
+  printf "%${terminal_width:-80}s\n" | tr ' ' '#'
+}
+
+function echo_running_sudo() {
+  if [ "$(id -u -n)" == "root" ] ; then
+    echo "Running $1"
+  else
+    echo "Running sudo $1; you may be prompted for your password."
+  fi
+}
+
+#-------------------------------------------------------------------------------
+# Misc helper functions.
+
+# Execute the args (a command) as root, either using sudo or directly if already
+# running as root.
+function my_sudo() {
+  if [ "$(id -u -n)" == "root" ] ; then
```

Comment:
If one is root then does running it with sudo not work? Just wondering why we can't always use sudo regardless of user. For Docker I just install sudo to make it work with default root user.

Reply:
Inside of docker sudo is not normally available. And it is recommended to NOT install sudo or su.